sitestats/sitestats-impl/src/java/org/sakaiproject/sitestats/impl/StatsUpdateManagerImpl.java

```diff
@@ -706,11 +706,13 @@ private void consolidateEvent(Date dateTime, String eventId, String resourceRef,
 			}
 		} else if (eventId.startsWith(StatsManager.LESSONS_EVENTID_PREFIX)) {
 			String[] resourceParts = resourceRef.split("/");
-			if (resourceParts.length > 3 && "page".equals(resourceParts[2])) {
+			if (resourceParts.length > 3) {
+				//The references are something like "/lessonbuilder/page/4" the id is in the fourth position
 				long pageId = Long.parseLong(resourceParts[3]);
```

Comment:
Was the && "page".equals(resourceParts[2] protecting this from a possible NumberFormatException? Is it possible that an exception could exist here now if this value is not numeric? Maybe a check just in case?

Reply:
No, the comparison discards the /item/ elements and only processes the /page/ elements, since LB differentiates between item and pages, it should process both events.